src/Serilog.Sinks.Amazon.Kinesis/Stream/Sinks/HttpLogShipper.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright 2014 Serilog Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Amazon.Kinesis;
+using Amazon.Kinesis.Model;
+using Serilog.Sinks.Amazon.Kinesis.Logging;
+
+namespace Serilog.Sinks.Amazon.Kinesis.Stream.Sinks
+{
+    internal class HttpLogShipper : HttpLogShipperBase<PutRecordsRequestEntry, PutRecordsResponse>
+    {
+        readonly IAmazonKinesis _kinesisClient;
+
+        public HttpLogShipper(KinesisSinkState state) : base(state)
+        {
+            _kinesisClient = state.KinesisClient;
+        }
+
+        protected override PutRecordsRequestEntry PrepareRecord(byte[] bytes)
+        {
+            return new PutRecordsRequestEntry
+            {
+                PartitionKey = Guid.NewGuid().ToString(),
+                Data = new MemoryStream(bytes)
+            };
+        }
+
+        protected override PutRecordsResponse SendRecords(List<PutRecordsRequestEntry> records, out bool successful)
+        {
+            var request = new PutRecordsRequest
+            {
+                StreamName = _streamName,
+                Records = records
+            };
+
+            Logger.TraceFormat("Writing {0} records to kinesis", records.Count);
+            var response = _kinesisClient.PutRecords(request);
+
+            successful = response.FailedRecordCount == 0;
```

Comment:
An idea for the future, but it would be nice to start dealing with failed records properly, i.e. to resend them, not the whole batch. We could return a collection of failed records here (which could also replace the `out bool` parameter).